app/src/androidTest/java/com/github/fribourgsdp/radio/WorkingLobbyActivityTest.kt

```diff
@@ -166,46 +151,72 @@ class WorkingLobbyActivityTest {
         // Init views
         val uuidTextView = Espresso.onView(withId(R.id.uuidText))
-
         val testIntent = Intent(ctx, WorkingLobbyActivity::class.java).apply {
             putExtra(GAME_UID_KEY, -1)
             putExtra(GAME_IS_HOST_KEY, false)
         }
-
         ActivityScenario.launch<WorkingLobbyActivity>(testIntent).use {
             uuidTextView.check(matches(withText(ctx.getString(R.string.uid_error_join))))
         }
     }
-
     @Test
     fun lobbyFetchesLyricsTest() {
         val lyrics = "If you feel, little chance, make a stance\n" +
-            "Looking for, better days, let me say\n" +
-            "Something's wrong, when you can't, let me go\n" +
-            "For to long, long, long...\n" +
-            "\n" +
-            "Momentum owns you\n" +
-            "Controlling her too"
+                "Looking for, better days, let me say\n" +
+                "Something's wrong, when you can't, let me go\n" +
+                "For to long, long, long...\n" +
+                "\n" +
+                "Momentum owns you\n" +
+                "Controlling her too"
         // Test values
         var testHost = User("Test User")
-
         val testPlaylist = Playlist("Stoner Playlist")
         testPlaylist.addSong(Song("Momentum", "Truckfighters"))
         testHost.addPlaylist(testPlaylist)
-
         val context: Context = ApplicationProvider.getApplicationContext()
         testHost.save(context)
-
         val testIntent = Intent(ctx, LyricsGettingWorkingLobbyActivity::class.java)
-
         ActivityScenario.launch<LyricsGettingWorkingLobbyActivity>(testIntent).use {
             testHost = User.load(context)
             assertTrue(testHost.getPlaylists().any { p -> p.getSongs().any { s -> s.lyrics == lyrics } })
         }
     }
+
+    @Test
+    fun createCorrectQrCode(){
+        // Test values
+        val testName = "Hello World!"
+        val testPlaylist = Playlist("Rap Playlist")
+        val testNbRounds = 20
+        val withHint = true
+        val private = true
+        val intent = Intent(ctx, WorkingLobbyActivity::class.java).apply {
+            putExtra(GAME_NAME_KEY, testName)
+            putExtra(GAME_PLAYLIST_KEY, Json.encodeToString(testPlaylist))
+            putExtra(GAME_NB_ROUNDS_KEY, testNbRounds)
+            putExtra(GAME_HINT_KEY, withHint)
+            putExtra(GAME_PRIVACY_KEY, private)
+            putExtra(GAME_IS_HOST_KEY, true)
+        }
+        ActivityScenario.launch<WorkingLobbyActivity>(intent).use { scenario ->
+
+
+            val displayQRCodeButton = Espresso.onView(withId(R.id.displayQRCodeButton))
+            displayQRCodeButton.perform(ViewActions.click())
+
+            Espresso.onView(withId(R.id.cancel_button))
+                .inRoot(isDialog())
+                .check(matches(isDisplayed()))
+
+            Espresso.onView(withId(R.id.cancel_button))
+                .inRoot(isDialog())
+                .perform(ViewActions.click())
```

Comment:
Same remark as above, don't you want to test what it does after cancelling?

Reply:
yes indeed an oversight on my part, well done